### PR TITLE
chore (pkg/scale): refactor the types package to utilize new scale package

### PR DESCRIPTION
**Pull request by jimjbrettj**
## Changes

- Wrote new types for digest, consensusDigest, header, blockData, and block that are compatible with the new scale package
- Integrated scale and new types into dot/digest and dot/network
- Finished the integration of the new block type, so in this PR I am looking specifically for feedback on dot/digest and dot/network integration, the new block type, as well as the other new types that are still a work in progress. Also am open to suggestions on changes to my approach to this integration
- I created a temporary branch to merge into as a staging branch before I merge into dev so that I can get more steady feedback on my work, as well as hopefully make the reviewing process a little easier given the size of this refactor

## Tests

```
- go test ./dot/types -v -short
- go test ./dot/network -v -short
- go test ./dot/digest -v -short
```

## Issues

- #1745 
- #1744 
- #1743 

## Primary Reviewer 

- @timwu20

### CODE REVIEW

File: dot/core/interface.go [hunk 1]
```diff
@@ -58,7 +60,8 @@ type StorageState interface {
 	LoadCode(root *common.Hash) ([]byte, error)
 	LoadCodeHash(root *common.Hash) (common.Hash, error)
 	TrieState(root *common.Hash) (*rtstorage.TrieState, error)
-	StoreTrie(*rtstorage.TrieState, *types.Header) error
+	//StoreTrie(*rtstorage.TrieState, *types.Header) error
+	StoreTrieVdt(*rtstorage.TrieState, *types.HeaderVdt) error
```
Comment (timwu20) [inline dot/core/interface.go:64]: Are you going to replace `type.Header` and revert to the original function names?

Comment (jimjbrettj) [reply dot/core/interface.go:64]: Yup exactly. Anything with "new" or "vdt" appended to the name is temporary and will be changed back to the original naming after the integration.

Comment (timwu20) [reply dot/core/interface.go:64]: can you do the renames I think it'll make the diff look much nicer.

Comment (jimjbrettj) [reply dot/core/interface.go:64]: Absolutely I agree. I am actually working on this currently

**Pull request closed without merging**